src/Widgets/Tooltip.vala

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2021 elementary, Inc (https://elementary.io)
+ *           2021 José Expósito <jose.exposito89@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Clutter actor to display text in a tooltip-like component.
+ */
+public class Gala.Tooltip : Clutter.Actor {
+    private static Clutter.Color text_color;
+    private static Gdk.RGBA bg_color;
+    private static Gtk.Border padding;
+    private static int border_radius;
+
+    /**
+     * Canvas to draw the Tooltip background.
+     */
+    private Clutter.Canvas background_canvas;
+
+    /**
+     * Actor to display the Tooltip text.
+     */
+    private Clutter.Text? text_actor = null;
+
+    /**
+     * Text displayed in the Tooltip.
+     * @see set_text
+     */
+    private string text;
+
+    /**
+     * Maximum width of the Tooltip.
+     * @see set_max_width
+     */
+    public float max_width;
+
+    static construct {
+        var dummy_label = new Gtk.Label ("") {
+            tooltip_text = "null"
+        };
+
+        unowned var label_style_context = dummy_label.get_style_context ();
+
+        var widget_path = label_style_context.get_path ().copy ();
+        widget_path.iter_set_object_name (-1, "tooltip");
```

Comment:
I know this has been done before like this in gala, but I'm pretty sure the path itself can be constructed without initializing a new label. This is fine though.

Reply:
Thanks for pushing me in that direction! fixed